sdks/java/io/bigquery-io-tests/src/test/java/org/apache/beam/sdk/bigqueryiotests/BigQueryIOIT.java

```diff
@@ -113,33 +120,34 @@ public static void setup() throws IOException {
         SyntheticOptions.fromJsonString(options.getSourceOptions(), SyntheticSourceOptions.class);
     bigQueryMetricsDataset = options.getBigQueryDataset();
     bigQueryMetricsTable = options.getBigQueryTable();
-    bigQuerySourceDataset = options.getBigQuerySourceDataset();
-    bigQuerySourceTable = options.getBigQuerySourceTable();
+    bigQueryTestDataset = options.getBigQueryTestDataset();
+    bigQueryTestTable = options.getBigQueryTestTable();
+    useStreamingWrites = options.getUseStreamingWrites();
     BigQueryOptions bigQueryOptions = BigQueryOptions.newBuilder().build();
     tableQualifier =
         String.format(
-            "%s:%s.%s", bigQueryOptions.getProjectId(), bigQuerySourceDataset, bigQuerySourceTable);
-    if (!checkForSourceTable()) {
-      createAndFillSourceTable();
-    }
+            "%s:%s.%s", bigQueryOptions.getProjectId(), bigQueryTestDataset, bigQueryTestTable);
   }
 
   @AfterClass
   public static void tearDown() {
     BigQueryOptions options = BigQueryOptions.newBuilder().build();
     BigQuery client = options.getService();
-    TableId tableId =
-        TableId.of(options.getProjectId(), bigQuerySourceDataset, bigQuerySourceTable);
+    TableId tableId = TableId.of(options.getProjectId(), bigQueryTestDataset, bigQueryTestTable);
     client.delete(tableId);
   }
 
-  private static void createAndFillSourceTable() {
-
-    Pipeline pipeline = Pipeline.create();
+  private void testWrite() {
+    Pipeline pipeline = Pipeline.create(options);
 
+    BigQueryIO.Write.Method method =
+        useStreamingWrites
+            ? BigQueryIO.Write.Method.STREAMING_INSERTS
+            : BigQueryIO.Write.Method.FILE_LOADS;
     pipeline
         .apply("Read from source", Read.from(new SyntheticBoundedSource(sourceOptions)))
         .apply("Map records", ParDo.of(new MapKVToV()))
+        .apply("Gather time", ParDo.of(new TimeMonitor<>(NAMESPACE, WRITE_TIME_METRIC_NAME)))
```

Comment:
Why not place the monitor directly after Read?

Reply:
An oversight, I fixed it